src/realm/sync/noinst/client_impl_base.cpp

```diff
@@ -1861,6 +1859,7 @@ void Session::send_bind_message()
         if (auto migrated_partition = get_migration_store()->get_migrated_partition()) {
             bind_json_data["migratedPartition"] = *migrated_partition;
         }
+        bind_json_data["schema_version"] = get_schema_version();
```

Comment:
I am assuming the server will treat a `-1` schema version the same as if the value was not provided.
Should we exclude the `schema_version` value if it is "not versioned"?
or maybe exclude it for now and always include it once we have tested core against the server schema versions feature?
If there are changes to core required to support the new feature, it could be used to advertise to the server that we support the feature.

Reply:
There are client changes required for this feature. One way to advertise the client supports the feature is by bumping the sync protocol version (we're still debating if it's really needed). My thinking was that once an app is upgraded to support the future, the client will always report a schema version even if there is none.

We can always stop reporting the schema version in some cases if we deem it necessary (this is all going to a feature branch).

Reply:
With the bind json data, I don't think you need to bump the proto version - hopefully, the presence of the value should be enough to indicate whether or not the client supports schema migrations (which was my concern if this was being merged to master). Although, this only works for FLX, and the proto version would need to be bumped if it also has to work for PBS.

Reply:
yeah, that's the idea. And that's why I think it would be a bit odd to sometimes not report it (that should only happen if the app is downgraded). But this may change once we start integration testing.